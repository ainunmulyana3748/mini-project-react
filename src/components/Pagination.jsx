import React from "react";
import Button from "./Button";

const Pagination = ({ currentPage, totalPages, onPageChange }) => {
  return (
    <div className="flex gap-4 justify-center mt-5">
      <Button
        onClick={() => onPageChange(currentPage - 1)}
        disabled={currentPage === 1}
        className="px-4 py-2 rounded bg-black text-white hover:bg-zinc-800 disabled:opacity-50 cursor-pointer"
      >
        Prev
      </Button>

      {[...Array(totalPages)].map((_, index) => {
        const pageNum = index + 1;
        return (
          <div
            key={pageNum}
            onClick={() => onPageChange(pageNum)}
            className={`w-[50px] rounded font-bold text-center p-2 cursor-pointer ${
              currentPage === pageNum
                ? "bg-gray-500 text-white"
                : "bg-black text-white hover:bg-zinc-700"
            }`}
          >
            {pageNum}
          </div>
        );
      })}

      <Button
        onClick={() => onPageChange(currentPage + 1)}
        disabled={currentPage === totalPages}
        className="px-4 py-2 rounded bg-black text-white hover:bg-zinc-800 disabled:opacity-50 cursor-pointer"
      >
        Next
      </Button>
    </div>
  );
};

export default Pagination;
