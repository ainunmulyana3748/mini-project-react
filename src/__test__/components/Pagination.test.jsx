import { render, screen, fireEvent } from "@testing-library/react";
import Pagination from "../../components/Pagination";

describe("Pagination Component", () => {
  const onPageChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("render tombol prev, next, dan nomor halaman", () => {
    render(
      <Pagination currentPage={2} totalPages={3} onPageChange={onPageChange} />
    );
    expect(screen.getByText("Prev")).toBeInTheDocument();
    expect(screen.getByText("Next")).toBeInTheDocument();
    expect(screen.getByText("1")).toBeInTheDocument();
    expect(screen.getByText("2")).toBeInTheDocument();
    expect(screen.getByText("3")).toBeInTheDocument();
  });

  test("klik prev, next, dan nomor halaman memanggil onPageChange", () => {
    render(
      <Pagination currentPage={2} totalPages={3} onPageChange={onPageChange} />
    );
    fireEvent.click(screen.getByText("Prev"));
    expect(onPageChange).toHaveBeenCalledWith(1);
    fireEvent.click(screen.getByText("Next"));
    expect(onPageChange).toHaveBeenCalledWith(3);
    fireEvent.click(screen.getByText("1"));
    expect(onPageChange).toHaveBeenCalledWith(1);
    fireEvent.click(screen.getByText("3"));
    expect(onPageChange).toHaveBeenCalledWith(3);
  });

  test("snapshot pagination", () => {
    const { container } = render(
      <Pagination currentPage={2} totalPages={3} onPageChange={onPageChange} />
    );
    expect(container).toMatchSnapshot();
  });
});
