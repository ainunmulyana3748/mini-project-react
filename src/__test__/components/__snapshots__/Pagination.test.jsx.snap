// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component snapshot pagination 1`] = `
<div>
  <div
    class="flex gap-4 justify-center mt-5"
  >
    <button
      class="px-4 py-2 rounded bg-black text-white hover:bg-zinc-800 disabled:opacity-50 cursor-pointer"
    >
      Prev
    </button>
    <div
      class="w-[50px] rounded font-bold text-center p-2 cursor-pointer bg-black text-white hover:bg-zinc-700"
    >
      1
    </div>
    <div
      class="w-[50px] rounded font-bold text-center p-2 cursor-pointer bg-gray-500 text-white"
    >
      2
    </div>
    <div
      class="w-[50px] rounded font-bold text-center p-2 cursor-pointer bg-black text-white hover:bg-zinc-700"
    >
      3
    </div>
    <button
      class="px-4 py-2 rounded bg-black text-white hover:bg-zinc-800 disabled:opacity-50 cursor-pointer"
    >
      Next
    </button>
  </div>
</div>
`;
